pkg/vault/test/foundry/BufferDoSProtection.t.sol

```diff
@@ -0,0 +1,415 @@
+        uint256 frontrunnerWrappedAmount
+    ) private view {
+        (uint256 daiIdx, uint256 waDaiIdx, IERC20[] memory tokens) = _getTokenArrayAndIndexesOfWaDaiBuffer();
+        BaseVaultTest.Balances memory balancesAfter = getBalances(lp, tokens);
+
+        // Check unwrap results.
+        assertEq(amountOut, _wrapAmount, "AmountOut (underlying withdrawn) is wrong");
+        assertEq(amountIn, waDAI.previewDeposit(_wrapAmount), "AmountIn (wrapped burned) is wrong");
+
+        // Check user balances.
+        assertEq(
+            balancesAfter.lpTokens[daiIdx],
+            balancesBefore.lpTokens[daiIdx] + _wrapAmount,
+            "LP balance of underlying token is wrong"
+        );
+        assertEq(
+            balancesAfter.lpTokens[waDaiIdx],
+            balancesBefore.lpTokens[waDaiIdx] - waDAI.previewWithdraw(_wrapAmount),
+            "LP balance of wrapped token is wrong"
+        );
+
+        // Check alice (frontrunner) balances. The frontrunner should lose the funds.
+        assertEq(
+            balancesAfter.aliceTokens[daiIdx],
+            balancesBefore.aliceTokens[daiIdx],
+            "Frontrunner balance of underlying token is wrong"
+        );
+        assertEq(
+            balancesAfter.aliceTokens[waDaiIdx],
+            balancesBefore.aliceTokens[waDaiIdx] - frontrunnerWrappedAmount,
+            "Frontrunner balance of wrapped token is wrong"
+        );
+
+        // Check Vault reserves. Vault should have the reserves from before and the frontrunner amount (the user paid
+        // for the amount deposited into the wrapper protocol, so the vault reserves should not be affected by that).
+        assertEq(
+            balancesAfter.vaultReserves[daiIdx],
+            balancesBefore.vaultReserves[daiIdx],
+            "Vault reserves of underlying token is wrong"
+        );
+        assertEq(
+            balancesAfter.vaultReserves[waDaiIdx],
+            balancesBefore.vaultReserves[waDaiIdx] + frontrunnerWrappedAmount,
+            "Vault reserves of wrapped token is wrong"
+        );
+
+        // Check Vault balances. Vault balances should match vault reserves.
+        assertEq(
+            balancesAfter.vaultReserves[daiIdx],
+            balancesAfter.vaultTokens[daiIdx],
+            "Vault balance of underlying token is wrong"
+        );
+        assertEq(
+            balancesAfter.vaultReserves[waDaiIdx],
+            balancesAfter.vaultTokens[waDaiIdx],
+            "Vault balance of wrapped token is wrong"
+        );
+    }
+
+    function _getTokenArrayAndIndexesOfWaDaiBuffer()
+        private
+        view
+        returns (uint256 daiIdx, uint256 waDaiIdx, IERC20[] memory tokens)
+    {
+        (daiIdx, waDaiIdx) = getSortedIndexes(address(dai), address(waDAI));
+        tokens = new IERC20[](2);
+        tokens[daiIdx] = dai;
+        tokens[waDaiIdx] = IERC20(address(waDAI));
+    }
+
+    /// @notice Hook used to interact with ERC4626 wrap/unwrap primitive of the vault.
```

Comment:
```suggestion
    /// @notice Hook used to interact with the ERC4626 wrap/unwrap primitive of the Vault.
```
Technically should capitalize Vault here too, but less important for tests. Not sure that's consistent everywhere.